job/server/src/main/java/alluxio/worker/job/task/PausableThreadPoolExecutor.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.worker.job.task;
+
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.locks.Condition;
+import java.util.concurrent.locks.ReentrantLock;
+
+/**
+ * ThreadPoolExecutor that can be temporarily paused to prevent any threads from
+ * executing new tasks.
+ */
+public class PausableThreadPoolExecutor extends ThreadPoolExecutor {
+
+  private boolean mIsPaused;
+  private int mNumPaused;
```

Comment:
can you comment on these variables? Especially, I think access has to be protected by the `mPauseLock`?